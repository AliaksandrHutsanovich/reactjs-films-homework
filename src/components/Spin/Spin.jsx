import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Spin as AntdSpin } from 'antd';

import '!style-loader!css-loader!sass-loader!antd/dist/antd.css'; //eslint-disable-line

const Spin = ({ size }) => (
  <AntdSpin size={size} />
);

Spin.propTypes = {
  size: PropTypes.oneOf([
    'small',
    'default',
    'large',
  ]),
};

Spin.defaultProps = {
  size: 'default',
};

export default memo(Spin);
