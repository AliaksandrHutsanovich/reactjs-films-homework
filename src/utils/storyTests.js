const storyTests = (stories, renderer) => {
  Object.keys(stories).forEach((storyKey) => {
    if (typeof stories[storyKey] !== 'function') {
      return;
    }
    it(`${storyKey} should be rendered properly`, () => {
      renderer.render(stories[storyKey]());
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
};

export default storyTests;
