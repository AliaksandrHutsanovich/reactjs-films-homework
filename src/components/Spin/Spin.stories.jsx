import React from 'react';
import Spin from './Spin';

export default {
  component: Spin,
  title: 'components/Spin',
};

export const defaultSpin = () => (
  <Spin />
);

export const smallSpin = () => (
  <Spin size="small" />
);

export const largeSpin = () => (
  <Spin size="large" />
);
