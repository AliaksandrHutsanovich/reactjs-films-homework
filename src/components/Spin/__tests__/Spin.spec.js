import ShallowRenderer from 'react-test-renderer/shallow';
import storyTests from '../../../utils/storyTests';
import * as stories from '../Spin.stories';

describe('<Spin />', () => {
  const renderer = new ShallowRenderer();

  storyTests(stories, renderer);
});
