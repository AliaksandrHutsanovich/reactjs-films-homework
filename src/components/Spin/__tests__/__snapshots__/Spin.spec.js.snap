// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spin /> defaultSpin should be rendered properly 1`] = `
<Spin
  size="default"
  spinning={true}
  wrapperClassName=""
/>
`;

exports[`<Spin /> largeSpin should be rendered properly 1`] = `
<Spin
  size="large"
  spinning={true}
  wrapperClassName=""
/>
`;

exports[`<Spin /> smallSpin should be rendered properly 1`] = `
<Spin
  size="small"
  spinning={true}
  wrapperClassName=""
/>
`;
